# Quantitative Analysis of Stock Market

#### importing libraries

In [1]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.io as pio
pio.templates.default = "plotly_white"

#### Load the dataset

In [2]:
stocks_data = pd.read_csv("stocks.csv")

#### Display the first few rows of the dataset

In [3]:
stocks_data.head()

Ticker        Date        Open        High         Low       Close  \
0   AAPL  2023-02-07  150.639999  155.229996  150.639999  154.649994   
1   AAPL  2023-02-08  153.880005  154.580002  151.169998  151.919998   
2   AAPL  2023-02-09  153.779999  154.330002  150.419998  150.869995   
3   AAPL  2023-02-10  149.460007  151.339996  149.220001  151.009995   
4   AAPL  2023-02-13  150.949997  154.259995  150.919998  153.850006   

    Adj Close    Volume  
0  154.414230  83322600  
1  151.688400  64120100  
2  150.639999  56007100  
3  151.009995  57450700  
4  153.850006  62199000

#### The dataset contains the following columns for stock market data:

- Ticker: The stock ticker symbol.
- Date: The trading date.
- Open: The opening price of the stock for the day.
- High: The highest price of the stock during the day.
- Low: The lowest price of the stock during the day.
- Close: The closing price of the stock for the day.
- Adj Close: The adjusted closing price, which accounts for all corporate actions such as dividends, stock splits, etc.
- Volume: The number of shares traded during the day.

####  Descriptive Statistics for each stock
##### Descriptive Statistics will provide summary statistics for each stock in the dataset. We’ll look at measures such as mean, median, standard deviation, and more for the Close prices:

In [4]:
descriptive_stats = stocks_data.groupby('Ticker')['Close'].describe()

print(descriptive_stats)

        count        mean        std         min         25%         50%  \
Ticker                                                                     
AAPL     62.0  158.240645   7.360485  145.309998  152.077499  158.055000   
GOOG     62.0  100.631532   6.279464   89.349998   94.702501  102.759998   
MSFT     62.0  275.039839  17.676231  246.270004  258.742500  275.810013   
NFLX     62.0  327.614677  18.554419  292.760010  315.672493  325.600006   

               75%         max  
Ticker                          
AAPL    165.162506  173.570007  
GOOG    105.962503  109.459999  
MSFT    287.217506  310.649994  
NFLX    338.899994  366.829987  


#### Let’s break down the results for each stock:

#### 1. AAPL (Apple Inc.)
- Count: 62.0 (The number of observations or trading days included in the dataset for AAPL)
- Mean: 158.24 (The average closing price)
- Standard Deviation: 7.36 (Measures the amount of variation or dispersion of closing prices)
- Minimum: 145.31 (The lowest closing price in the dataset)
- 25th Percentile: 152.08 (25% of the closing prices are below this value)
- Median (50%): 158.06 (The middle value of the closing prices)
- 75th Percentile: 165.16 (75% of the closing prices are below this value)
- Maximum: 173.57 (The highest closing price in the dataset)

#### 2. GOOG (Alphabet Inc.)
- Similar statistics as AAPL, but for GOOG. The mean closing price is 100.63, with a standard deviation of 6.28, indicating less variability in closing prices compared to AAPL.

#### 3. MSFT (Microsoft Corporation)
- The dataset includes the same number of observations for MSFT. It has a higher mean closing price of 275.04 and a higher standard deviation of 17.68, suggesting greater price variability than AAPL and GOOG.

#### 4.NFLX (Netflix Inc.)
- NFLX shows the highest mean closing price (327.61) among these stocks and the highest standard deviation (18.55), indicating the most significant price fluctuation.

## Time Series Analysis
Next, we’ll proceed with the Time Series Analysis to examine trends and patterns over time, focusing on the closing prices:

In [5]:
stocks_data['Date'] = pd.to_datetime(stocks_data['Date'])
pivot_data = stocks_data.pivot(index='Date', columns='Ticker', values='Close')

In [6]:
fig = make_subplots(rows=1, cols=1)

In [7]:
for column in pivot_data.columns:
    fig.add_trace(
        go.Scatter(x=pivot_data.index, y=pivot_data[column], name=column),
        row=1, col=1
    )

In [11]:
fig.update_layout(
    title_text='Time Series of Closing Prices',
    xaxis_title='Date',
    yaxis_title='Closing Price',
    legend_title='Ticker',
    showlegend=True
)

fig.show()